catalog/mvcc/validate.go

```diff
@@ -100,7 +102,11 @@ func ValidateCommitter(name string) ValidateFunc {
 
 func ValidateStorageNamespace(storageNamespace string) ValidateFunc {
 	return func() bool {
-		return IsNonEmptyString(storageNamespace)
+		if !IsNonEmptyString(storageNamespace) {
+			return false
+		}
+		configuredBlockStoreType := viper.GetString("blockstore.type")
```

Comment:
I think that this code should go in `api.Controller.CreateRepositoryHandler`.  There you have `deps`, and you could ask `deps.BlockAdapter` whether it likes this storage namespace.

Why?
We prefer not to read configuration from arbitrary locations, it creates tricky dependencies (e.g. if `blockstore.type` is renamed, or if we add another way to configure blockstores).  It is OK for catalog to validate storage namespace for generally being nonempty, but catalog has no idea of the blockstore and should not do this validation.

Reply:
Ok. I will revert it back to original.
As far as `api_controller` is concerned, I think it is already done.
https://github.com/treeverse/lakeFS/blob/8731b60fb472e28ac714933aff0c8ee3bc5b14de/api/api_controller.go#L547

Reply:
Thanks, I missed that!  It's a bit subtle, but since `block.Adapter.Put` actually verifies the type of the storage namespace with the adaptor everything works.